lang/java/reef-common/src/main/java/org/apache/reef/driver/restart/EvaluatorRestartState.java

```diff
@@ -23,22 +23,53 @@
 import org.apache.reef.annotations.audience.Private;
 
 /**
- * A static utilities class for simplifying calls to driver restart manager.
+ * The state that the evaluator is in in the driver restart process.
  */
 @Private
 @DriverSide
 @Unstable
-public final class DriverRestartUtilities {
+public enum EvaluatorRestartState {
```

Comment:
Maybe add a static method `boolean isLegalTransition(EvaluatorRestartState old, EvaluatorRestartState new)` to check transitions between states.